src/components/modal/modal.tsx

```diff
@@ -355,11 +346,13 @@ export class Modal implements ConditionalSlotComponent, OpenCloseComponent {
    */
   @Method()
   async setFocus(focusId?: "close-button"): Promise<void> {
-    const closeButton = this.closeButtonEl;
+    const { closeButtonEl } = this;
 
-    return focusElement(
-      focusId === "close-button" ? closeButton : getFocusableElements(this.el)[0] || closeButton
-    );
+    if (closeButtonEl && focusId === "close-button") {
```

Comment:
Since `close-button` is the only supported ID, I think you could just check for `closeButtonEl` being present or not. Passing any other ID would technically be invalid, right?

Reply:
Correct. I think its nicer to check and fallback incase the user entered an invalid ID>